internal/acceptance_test/resource_host_test.go

```diff
@@ -178,9 +169,14 @@ func testAccCheckHostDestroy(t *testing.T, s *terraform.State) error {
 
 		ctx := p.GetContext()
 
-		host, _, err := p.Client.HostsApi.GetByID(ctx, hostID)
-		if err == nil && host.State != rest.HOSTSTATE_DELETED {
-			return fmt.Errorf("Host: %s still exists", hostID)
+		_, resp, err := p.Client.HostsApi.GetByID(ctx, hostID)
+		if err != nil {
+			return fmt.Errorf("Error retrieving host %s: %v", hostID, err)
+		}
+		defer resp.Body.Close()
```

Comment:
Same here, not sure about deferring in a loop. resp.Body doesn't look to be used, so just close it.